### PR TITLE
Clean up logging options

### LINKED ISSUE #1187: Clean up logging options
These are user interface bugs:

- [ ] `--log none` makes no sense; the user should not be able to suppress error reporting.
- [ ] The default `--log-level` should be `warning`.
- [ ] `--log stdout` should be `--log stderr`.
- [ ] Fix the `--help` to explain how to use `--log` to specifying a filename.

### PR DESCRIPTION
**Pull request by andreiburdusa**
---
Fixes #1187 
First part: #1213 
- Fix the `--help` to explain how to use `--log` to specifying a filename.
###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

File: kore/src/Kore/Repl/Data.hs [hunk 1]
```diff
@@ -351,8 +351,9 @@ helpText =
     \                                      these scopes are used for filtering\
                                            \ the logged information, for example,\
                                            \ '[]' will log all scopes\n\
-    \                                      <type> can be 'stderr' or\n\
-                                           \'file filename'\n\
+    \                                      <type> can be 'stderr' or 'filetext';\
+                                           \ the latter prints the output to\
+                                           \ \"./kore-(date).log\"\n\
```
Comment (ttuegel) [inline kore/src/Kore/Repl/Data.hs]: Is this correct, that `kore-repl` has a hard-coded filename for the log? What about `kore-exec`?

Comment (ana-pantilie) [reply kore/src/Kore/Repl/Data.hs]: It shouldn't be hard-coded in `kore-repl`

**Comment by vladciobanu** [comment]
What I meant during the meeting: for the `repl`, the log file is not hardcoded anywhere. See https://github.com/kframework/kore/blob/master/kore/src/Kore/Repl/Data.hs#L179 and its parser https://github.com/kframework/kore/blob/master/kore/src/Kore/Repl/Parser.hs#L281 so the text change is wrong: you can specify the filename in the logger.

However, for `kore-exec`, the file name is hardcoded here: https://github.com/kframework/kore/blob/master/kore/src/Kore/Logger/Output.hs#L137 and the help text for the log flag is here https://github.com/kframework/kore/blob/master/kore/src/Kore/Logger/Output.hs#L157

Maybe what needs to be done here is just to change this type here https://github.com/kframework/kore/blob/master/kore/src/Kore/Logger/Output.hs#L99 to add a FilePath field to the second ctor and update the code accordingly...?

File: kore/src/Kore/Logger/Output.hs [hunk 2]
```diff
@@ -95,9 +95,9 @@ import Kore.Logger
 -- the logger will operate.
 data KoreLogType
     = LogStdErr
-    -- ^ log to StdErr when '--log StdErr' is passed
-    | LogFileText
-    -- ^ log to "./kore-(date).log" when '--log FileText' is passed
+    -- ^ log to StdErr when '--log stderr' is passed
```
Comment (ttuegel) [inline kore/src/Kore/Logger/Output.hs]:
```suggestion
    -- ^ log to stderr (default)
```

**Pull request merged**